File: 3_yolo_python_standard_msg/yolo_python_standard_msg/sub.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import Int32

class VisionSubscriber(Node):
    def __init__(self):
        super().__init__('vision_sub_node')

        self.bounding_box = 0
        self.titik_tengah = 0
        self.fps = 0

        self.sub_bbox = self.create_subscription(
            Int32,
            'bounding_box_topic',
            self.bbox_callback,
            10)

        self.sub_center = self.create_subscription(
            Int32,
            'center_topic',
            self.center_callback,
            10)

        self.sub_fps = self.create_subscription(
            Int32,
            'fps_topic',
            self.fps_callback,
            10)

    def bbox_callback(self, msg):
        self.bounding_box = msg.data
        self.print_data()

    def center_callback(self, msg):
        self.titik_tengah = msg.data
        self.print_data()

    def fps_callback(self, msg):
        self.fps = msg.data
        self.print_data()

    def print_data(self):
        # Print all data whenever any callback received new data
        self.get_logger().info(f'Bounding Box : {self.bounding_box}')
        self.get_logger().info(f'Titik Tengah : {self.titik_tengah}')
        self.get_logger().info(f'FPS         : {self.fps}')
        self.get_logger().info('-----------------------------')

def main(args=None):
    rclpy.init(args=args)
    node = VisionSubscriber()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: 4_yolo_cpp_standard_msg/src/sub.cpp
#include <chrono>
#include <memory>
#include <iostream>
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/int32.hpp"

using std::placeholders::_1;

class VisionSubscriber : public rclcpp::Node
{
public:
  VisionSubscriber()
  : Node("vision_sub_node")
  {
    sub_bbox_ = this->create_subscription<std_msgs::msg::Int32>(
      "bounding_box_topic", 10, std::bind(&VisionSubscriber::bbox_callback, this, _1));

    sub_center_ = this->create_subscription<std_msgs::msg::Int32>(
      "center_topic", 10, std::bind(&VisionSubscriber::center_callback, this, _1));

    sub_fps_ = this->create_subscription<std_msgs::msg::Int32>(
      "fps_topic", 10, std::bind(&VisionSubscriber::fps_callback, this, _1));
  }

private:
  void bbox_callback(const std_msgs::msg::Int32::SharedPtr msg)
  {
    std::cout << "Bounding Box : " << msg->data << std::endl;
  }

  void center_callback(const std_msgs::msg::Int32::SharedPtr msg)
  {
    std::cout << "Titik Tengah : " << msg->data << std::endl;
  }

  void fps_callback(const std_msgs::msg::Int32::SharedPtr msg)
  {
    std::cout << "FPS          : " << msg->data << std::endl;
    std::cout << "-----------------------------" << std::endl;
  }

  rclcpp::Subscription<std_msgs::msg::Int32>::SharedPtr sub_bbox_;
  rclcpp::Subscription<std_msgs::msg::Int32>::SharedPtr sub_center_;
  rclcpp::Subscription<std_msgs::msg::Int32>::SharedPtr sub_fps_;
};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<VisionSubscriber>());
  rclcpp::shutdown();
  return 0;
}
